Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/ColliderTranslationManipulators.cpp

```diff
@@ -103,55 +99,54 @@ namespace EMotionFX
         }
-
-        m_translationManipulators.SetLocalPosition(
-            m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position);
     }
 
-    void PhysicsSetupColliderTranslationManipulators::Teardown()
+    void ColliderTranslationManipulators::Teardown()
     {
+        PhysicsSetupManipulatorRequestBus::Handler::BusDisconnect();
         m_translationManipulators.Unregister();
     }
 
-    void PhysicsSetupColliderTranslationManipulators::ResetValues()
+    void ColliderTranslationManipulators::ResetValues()
     {
-        if (!m_physicsSetupManipulatorData.m_valid || !m_physicsSetupManipulatorData.m_colliderNodeConfiguration ||
-            m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes.empty())
+        if (m_physicsSetupManipulatorData.HasColliders())
         {
-            return;
+            BeginEditing(
+                m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position, AZ::Vector3::CreateZero());
+            FinishEditing(AZ::Vector3::CreateZero(), AZ::Vector3::CreateZero());
+            Refresh();
         }
-        m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position = AZ::Vector3::CreateZero();
-        m_translationManipulators.SetLocalPosition(AZ::Vector3::CreateZero());
     }
 
-    AZ::Vector3 PhysicsSetupColliderTranslationManipulators::GetPosition(const AZ::Vector3& startPosition, const AZ::Vector3& offset) const
+    AZ::Vector3 ColliderTranslationManipulators::GetPosition(const AZ::Vector3& startPosition, const AZ::Vector3& offset) const
     {
         const float scale = AZ::GetMax(AZ::MinTransformScale, m_physicsSetupManipulatorData.m_nodeWorldTransform.GetUniformScale());
         return startPosition + offset / scale;
     }
 
-    void PhysicsSetupColliderTranslationManipulators::OnManipulatorMoved(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
+    void ColliderTranslationManipulators::OnManipulatorMoved(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
     {
         AZ::Vector3 newPosition = GetPosition(startPosition, offset);
-        if (m_physicsSetupManipulatorData.m_valid && m_physicsSetupManipulatorData.m_colliderNodeConfiguration &&
-            !m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes.empty())
+        if (m_physicsSetupManipulatorData.HasColliders())
         {
             m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position = newPosition;
         }
         m_translationManipulators.SetLocalPosition(newPosition);
         m_physicsSetupManipulatorData.m_collidersWidget->Update();
     }
 
-    void PhysicsSetupColliderTranslationManipulators::BeginEditing(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
+    void ColliderTranslationManipulators::BeginEditing(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
```

Comment:
Good rename :+1:

(it does may be wonder if in future there is logic we could steal from the PhysX collider manipulators in the main editor... 🤔 Maybe something to look at after to see what similarities (if any) there are)

Reply:
I think it's a bit tricky because there are so many differences like expecting an entity component ID pair, and using a completely different undo system.